### PR TITLE
feat(VocalTrack): Add CRUD

### PR DESCRIPTION
**Pull request by Dextorr**
Add CRUD

### CODE REVIEW

File: src/api/client/endpoints/vocal_track.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+export class VocalTrackEndpoint {
+  public path = `${ENDPOINT.VOCAL_TRACKS}`
+  constructor(protected readonly client: IHttpClient) {}
+
+  /**
+   * This method allows you to create a new vocalTrack
+   * @param payload
+   */
+  public async create(payload: IVocalTrackCreate & ITrackAudioFiles): ApiResponse<IVocalTrack> {
```
Comment (useless-stuff) [inline src/api/client/endpoints/vocal_track.ts:12]: why we don't have 2 parameters?
one for the file and one for the data?
it's gonna be a clear type hinting

File: src/api/client/endpoints/vocal_track.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    const vocalTrackResponse = await this.client.post<IVocalTrackCreate, IVocalTrack>(this.path, vocalTrackCreate)
+    uploadFileWithAwsPolicy(vocalTrackResponse.data.uploadPolicy, originalTrackAudio)
+    return vocalTrackResponse
+  }
+
+  /**
+   * This method returns the vocalTrack model matching the vocalTrackId param
+   * @param vocalTrackId
+   */
+  public get(remixTrackId: string): ApiResponse<IVocalTrack> {
```
Comment (useless-stuff) [inline src/api/client/endpoints/vocal_track.ts:23]: remixTrackId

File: src/api/client/endpoints/vocal_track.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    return this.client.get<IVocalTrack>(`${this.path}/${remixTrackId}`)
```
Comment (useless-stuff) [inline src/api/client/endpoints/vocal_track.ts:24]: idem

File: src/api/client/endpoints/vocal_track.ts (hunk 1 above)
Comment (julian-aimusic) [reply src/api/client/endpoints/vocal_track.ts:12]: its the same way i did for backingTrack and @Gio85 for remixTrack 

if we want to have 2 parameters i think we should change those too

Comment (Dextorr) [reply src/api/client/endpoints/vocal_track.ts:12]: Exactly as Julian said, would you like me to change for all of them?

**Pull request merged**